### DISCUSSION TITLE
How to connect to external redis app

### DISCUSSION THREAD

**Discussion by snehlata08** [apache/superset#22909] [Q&A / Help]
I have postgres and redis containers running outside of the superset app. I want to connect those containers with supeset app in docker and helm. 
Superset starts its own postgres and redis container which i don't want. Can anyone show me the steps to integrate to external postgres and redis docker containers. 
I am using superset version 2.0

**Comment by msriaj** [comment]
You can integrate external PostgreSQL and Redis containers with Superset in the following steps:

Update the Superset configuration file (superset_config.py) with the appropriate connection details for the external PostgreSQL and Redis containers.

`SQLALCHEMY_DATABASE_URI = "postgresql://user:password@postgres_host:postgres_port/database_name"
CACHE_CONFIG = {'CACHE_TYPE': 'redis',
                'CACHE_REDIS_HOST': 'redis_host',
                'CACHE_REDIS_PORT': 'redis_port',
                'CACHE_REDIS_DB': 'redis_db',
                'CACHE_REDIS_URL': 'redis://redis_host:redis_port/redis_db'}
`

Update the Dockerfile for your Superset container to copy the updated configuration file into the container.

Build the updated Docker image for your Superset container.

Update the Helm chart for your Superset deployment to use the updated Docker image.

Deploy the updated Helm chart to connect the Superset container with the external PostgreSQL and Redis containers.

The above steps are just a general outline and may vary depending on your specific setup.
